programs/lz4io.c

```diff
@@ -395,8 +404,50 @@ typedef struct {
     void*  dstBuffer;
     size_t dstBufferSize;
     LZ4F_compressionContext_t ctx;
+    LZ4F_CDict* cdict;
 } cRess_t;
 
+static void LZ4IO_loadDict(char **dictionaryBuffer, size_t *dictionarySize) {
+    FILE* dictionaryFile;
+    size_t blockSize = 64 KB;
+    size_t dictionaryBufferSize = blockSize;
+    size_t readSize;
+    *dictionarySize = 0;
+    *dictionaryBuffer = malloc(dictionaryBufferSize);
```

Comment:
`malloc()` can fail. In which case `*dictionaryBuffer == NULL`.
Always check `malloc()` result, and act accordingly.